src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala

```diff
@@ -407,20 +417,29 @@ class StammdatenUpdateService(override val sysConfig: SystemConfig) extends Even
               tourlieferungMapping.column.sort -> None
             )
           case abo: Abo =>
-            // wechsel
-            val aboNeu = copyTo[HauptAbo, HeimlieferungAbo](abo, "vertriebId" -> va.vertriebId, "tourId" -> va.tourId, "tourName" -> tour.name, "vertriebsartId" -> update.vertriebsartIdNeu)
-            abo match {
-              case abo: DepotlieferungAbo => stammdatenWriteRepository.deleteEntity[DepotlieferungAbo, AboId](abo.id)
-              case abo: PostlieferungAbo  => stammdatenWriteRepository.deleteEntity[PostlieferungAbo, AboId](abo.id)
-              case _                      =>
-            }
-            stammdatenWriteRepository.insertEntity[HeimlieferungAbo, AboId](aboNeu)
+            stammdatenWriteRepository.getById(vertriebMapping, va.vertriebId) map { vertrieb =>
+              // wechsel
+              val aboNeu = copyTo[HauptAbo, HeimlieferungAbo](abo, "vertriebId" -> va.vertriebId, "tourId" -> va.tourId, "tourName" -> tour.name, "vertriebsartId" -> update.vertriebsartIdNeu, "vertriebBeschrieb" -> vertrieb.beschrieb)
+              abo match {
+                case abo: DepotlieferungAbo => stammdatenWriteRepository.deleteEntity[DepotlieferungAbo, AboId](abo.id)
+                case abo: PostlieferungAbo  => stammdatenWriteRepository.deleteEntity[PostlieferungAbo, AboId](abo.id)
+                case _                      =>
```

Comment:
Also strange for me, what's with TourlieferungAbo

Reply:
In case an abo changes the vertrieb, the type can change. So, it needs to be deleted from heim or tour and, for example, added to depot.